src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java

```diff
@@ -153,7 +157,8 @@ private ResponseEntity<XPayAuthResponse> createXpayAuthResponse(String errorMess
         }
 
         if (StringUtils.isEmpty(errorMessage)) {
-            String urlRedirect = StringUtils.join(responseUrlRedirect, requestId);
+            String sessionToken = jwtTokenUtils.generateToken(requestId);
+            String urlRedirect = StringUtils.join(responseUrlRedirect, requestId, "#token=", sessionToken);
```

Comment:
If you expect this strings to be not null prefer using `+` to concatenate strtings

Reply:
Done https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/210/commits/2ea78190d0495b5dfadfdafe1a469312ce6be9a0